app/controllers/tasks/collecting_events/parse/stepwise/lat_long_controller.rb

```diff
@@ -0,0 +1,210 @@
+      next_id        = current_collecting_event.id
+    end
+    # where do we go from here?
+    redirect_to collecting_event_lat_long_task_path(collecting_event_id: next_id,
+                                                    filters:             parse_filters(params))
+  end
+
+  def convert
+    lat                 = convert_params[:verbatim_latitude]
+    long                = convert_params[:verbatim_longitude]
+    retval              = {}
+    retval[:lat_piece]  = Utilities::Geo.degrees_minutes_seconds_to_decimal_degrees(lat)
+    retval[:long_piece] = Utilities::Geo.degrees_minutes_seconds_to_decimal_degrees(long)
+    render json: retval
+  end
+
+  def similar_labels
+    retval         = {}
+    lat            = similar_params[:lat]
+    long           = similar_params[:long]
+    piece          = similar_params[:piece]
+    # collecting_event_id = collecting_event_params[:collecting_event_id]
+    include_values = (similar_params[:include_values].nil?) ? false : true
+    sql_1          = '('
+    sql_1          += "verbatim_label LIKE '%#{sql_fix(lat)}%'" unless lat.blank?
+    sql_1          += " or verbatim_label LIKE '%#{sql_fix(long)}%'" unless long.blank?
+    sql_1          += " or verbatim_label LIKE '%#{sql_fix(piece)}%'" unless piece.blank?
+    sql_1          += ')'
+    sql_1          += ' and (verbatim_latitude is null or verbatim_longitude is null)' unless include_values
+    selected_items = CollectingEvent.where(sql_1)
+                       .with_project_id(sessions_current_project_id)
+                       .order(:id)
+                       .where.not(id: params[:collecting_event_id]).distinct
+
+    retval[:count] = selected_items.count.to_s
+    retval[:table] = render_to_string(partial: 'matching_table',
+                                      locals:  {
+                                        lat:            lat,
+                                        long:           long,
+                                        selected_items: selected_items
+                                      }) #([lat, long], selected_items)
+    render(json: retval)
+  end
+
+  protected
+
+  def sql_fix(item)
+    retval = item.gsub("'", "''")
+    retval
+  end
+
+  def collecting_event_id_param
+    retval = nil
+    begin
+      retval = params.require(:collecting_event_id)
+    rescue ActionController::ParameterMissing
+      # nothing provided, get the first possible one
+      # return CollectingEvent.with_project_id(sessions_current_project_id).order(:id).limit(1).pluck(:id)[0]
+      retval = Queries::CollectingEventLatLongExtractorQuery.new(
+        collecting_event_id: nil,
+        filters:             parse_filters(params))
+                 .all
+                 .with_project_id(sessions_current_project_id)
+                 .order(:id)
+                 .limit(1)
+                 .pluck(:id)[0]
+    end
+    retval
+  end
+
+  # TODO: deprecate for valud from view/helper
+  def next_collecting_event_id
```

Comment:
That's odd; you were the one who put it in the controller....
Also, I have no idea what that TODO was supposed to mean.